bundle/src/main/java/com/adobe/acs/commons/wcm/properties/shared/impl/SharedComponentPropertiesPageInfoProvider.java

```diff
@@ -87,4 +134,151 @@ public void updatePageInfo(SlingHttpServletRequest request, JSONObject info, Res
+                        break;
+
+                    default:
+                        break;
+                }
+            } catch (Exception e) {
+                log.error("Error determining if event affects list of components with shared/global properties", e);
+            }
+        }
+    }
+
+    /**
+     * Schedule an update of the map of components with shared/global properties to be updated
+     * 5 seconds from now.
+     *
+     * This handles race conditions where the map calculation happens before all nodes are installed,
+     * and also prevents stampedes from multiple JCR update events such as during a package installation.
+     */
+    private void scheduleSharedComponentsMapUpdate() {
+        Calendar timeToUpdate = Calendar.getInstance();
+        timeToUpdate.add(Calendar.SECOND, 5);
+        scheduledSharedComponentsMapUpdate = timeToUpdate;
+    }
+
+    /**
+     * Traverse the entire set of components in the /apps directory and create a map of all component types
+     * that have shared/global config dialogs.
+     *
+     * This is used by the JS libs in the authoring interface to determine if a component should show the
+     * options for editing shared/global configs.
+     */
+    private void updateSharedComponentsMap() {
+        ResourceResolver resourceResolver = null;
+        try {
+            log.debug("Calculating map of components with shared properties dialogs");
+
+            Map<String, Object> authInfo = Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, (Object) SERVICE_NAME);
+            resourceResolver = resourceResolverFactory.getServiceResourceResolver(authInfo);
+            ComponentManager componentManager = resourceResolver.adaptTo(ComponentManager.class);
+            Map<String, List<Boolean>> localComponentsWithSharedProperties = new HashMap<>();
+            for (Component component : componentManager.getComponents()) {
+                if (component.getPath().startsWith("/apps")) {
+                    boolean hasSharedDialogForTouch = componentHasTouchDialog(component, "dialogshared");
+                    boolean hasGlobalDialogForTouch = componentHasTouchDialog(component, "dialogglobal");
+                    boolean hasSharedDialogForClassic = componentHasClassicDialog(component, "dialog_shared");
+                    boolean hasGlobalDialogForClassic = componentHasClassicDialog(component, "dialog_global");
+                    if (hasSharedDialogForTouch || hasGlobalDialogForTouch || hasSharedDialogForClassic || hasGlobalDialogForClassic) {
+                        localComponentsWithSharedProperties.put(component.getResourceType(),
+                                Arrays.asList(hasSharedDialogForTouch, hasGlobalDialogForTouch, hasSharedDialogForClassic, hasGlobalDialogForClassic));
+                    }
+                }
+            }
+            componentsWithSharedProperties = localComponentsWithSharedProperties;
```

Comment:
this should be

    componentsWithSharedProperties = Collections.unmodifiableMap(localComponentsWithSharedProperties);